File: leetcode/0832. Flipping an Image/Solution.swift
// https://leetcode.com/problems/flipping-an-image

class Solution {
    func flipAndInvertImage(_ A: [[Int]]) -> [[Int]] {
        var result = [[Int]]()
        let n = A.count

        for row in A {
            var new_row = [Int]()
            for j in 0..<n {
                new_row.append(1 - row[n - j - 1])
            }
            result.append(new_row)
        }

        return result
    }
}

File: leetcode/0832. Flipping an Image/Solution.py
# https://leetcode.com/problems/flipping-an-image

from typing import List

class Solution:
    def flipAndInvertImage(self, A: List[List[int]]) -> List[List[int]]:
        result = []
        n = len(A)
        for row in A:
            new_row = []
            for j in range(n):
                new_row.append(1 - row[n - j - 1])
            result.append(new_row)
        
        return result

s = Solution()
assert s.flipAndInvertImage([[1,1,0],[1,0,1],[0,0,0]]) == [[1,0,0],[0,1,0],[1,1,1]]
assert s.flipAndInvertImage([[1,1,0,0],[1,0,0,1],[0,1,1,1],[1,0,1,0]]) == [[1,1,0,0],[0,1,1,0],[0,0,0,1],[1,0,1,0]]